File: backend/app/schemas/video.py
from pydantic import BaseModel
from typing import Optional
from enum import Enum

class VideoStatus(str, Enum):
    PENDING = "pending"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"

class VideoRequestCreate(BaseModel):
    query: str

class VideoRequestResponse(BaseModel):
    id: str
    query: str
    status: VideoStatus
    video_url: Optional[str] = None

    class Config:
        from_attributes = True

File: frontend_flutter/lib/models/video.dart
enum VideoStatus { pending, processing, completed, failed }

class VideoRequest {
  final String id;
  final String query;
  final VideoStatus status;
  final String? videoUrl;

  VideoRequest({
    required this.id,
    required this.query,
    required this.status,
    this.videoUrl,
  });

  factory VideoRequest.fromJson(Map<String, dynamic> json) {
    return VideoRequest(
      id: json['id'],
      query: json['query'],
      status: VideoStatus.values.firstWhere(
        (e) => e.toString().split('.').last == json['status'],
        orElse: () => VideoStatus.pending,
      ),
      // Ensure we get the full URL if the backend sends a relative path
      videoUrl: json['video_url'],
    );
  }
}